plugins/plugin-chart-table/src/transformProps.ts

```diff
@@ -98,11 +107,21 @@ export default function transformProps(chartProps: TableChartProps): TableChartT
           ...x,
           [key]:
             typeof time === 'string'
-              ? new Date(time.match(/T(\d{2}:){2}\d{2}$/) ? `${time}Z` : time)
+              ? new Date(
+                  time.match(REGEXP_TIMESTAMP_NO_TIMEZONE)
+                    ? // force UTC time for timestamps without a timezone
+                      `${time}Z`
+                    : time,
+                )
               : time,
         };
       });
-      formatter = getTimeFormatter(format || tableTimestampFormat, granularity);
+      // Use ganularity for "Adaptive Formatting" (smart_date)
+      const timeFormat = format || tableTimestampFormat;
+      formatter =
+        timeFormat === smartDateFormatter.id && isTimeColumn(key)
+          ? getTimeFormatterForGranularity(granularity)
+          : getTimeFormatter(timeFormat);
```

Comment:
@kristw I decided to not use time range formatter for this one because it doesn't seem necessary for table charts and it also produces ranges like "1999 - .999ms" when granularity is set to "year".

Reply:
sg